src/apps/backend/modules/task/internal/store/task-db.ts

```diff
@@ -6,6 +6,7 @@ export interface TaskDB {
   active: boolean;
   description: string;
   title: string;
+  sharedWith: Types.ObjectId[];
```

Comment:
This is not a good practice to store association as an array in the object itself. It makes it hard to query, reason, etc. You want to create a separate collection around this.

From an OOPS standpoint, this also feels somewhat unintuitive. A better way to model this would be via permission and who have access to what resource via permissions.

Ex: `TaskPermission`

This would have permission (READ | WRITE), account_id etc.